Kudu.Services/Deployment/PushDeploymentController.cs

```diff
@@ -138,9 +138,173 @@ public class PushDeploymentController : ApiController
+                            response.Content = new StringContent("JAR files cannot be deployed to " + website_framework + " web apps.");
+                            
+                            return response;
+                        }
+                        deploymentInfo.FileName = "app.jar";
+
+                        break;
+
+                    case Constants.Lib:
+                        if (string.IsNullOrWhiteSpace(path))
+                        {
+                            response.StatusCode = HttpStatusCode.BadRequest;
+                            response.Content = new StringContent("Path must be defined for library file deployments");
+                            
+                            return response;
+                        }
+                        else
+                        {
+
+                            //When the file name is unknown, then deploymentInfo.FileName must be derrived from the path query string
+                            deploymentInfo.FileName = Path.GetFileName(path);
+
+                            //If a path is provided, then deploymentInfo.TargetPath must also be derrived from the path query string
+                            deploymentInfo.TargetPath = Path.GetDirectoryName(path);
+                        }
+
+                        break;
+
+                    case Constants.Startup:
+
+                        //assumes windows application for Kudu deploymnets
+                        //deploymentInfo.TargetPath is set here because the files are not getting deployed to wwwwroot
+                        deploymentInfo.TargetPath = _environment.RootPath;
+                        deploymentInfo.FileName = "startup.bat";
+
+                        break;
+
+                    case Constants.Ear:
+
+                        //currently not supported on Windows but here for future use
+                        if (website_framework != Constants.JavaEE)
+                        {
+
+                            response.StatusCode = HttpStatusCode.BadRequest;
+                            response.Content = new StringContent("EAR files cannot be deployed to " + website_framework + " web apps.");
+                            
+                            return response;
+                        }
+                        deploymentInfo.FileName = "app.ear";
+
+                        break;
+
+                    case Constants.Static:
+                        if (string.IsNullOrWhiteSpace(path))
+                        {
+                            response.StatusCode = HttpStatusCode.BadRequest;
+                            response.Content = new StringContent("Path must be defined for static file deployments");
+                            
+                            return response;
+                        }
+                        else
+                        {
+                            deploymentInfo.FileName = Path.GetFileName(path);
```

Comment:
Looks like you expect this to be an absolute path? Can you add a few examples of valid and invalid paths and special cases.